scheduler/src/main/java/org/pentaho/platform/scheduler2/action/DefaultActionInvoker.java

```diff
@@ -108,25 +72,21 @@ protected IBackgroundExecutionStreamProvider getStreamProvider( final Map<String
   public IActionInvokeStatus invokeAction( final IAction actionBean,
                                            final String actionUser,
                                            final Map<String, Serializable> params ) throws Exception {
-    ActionUtil.prepareMap( params );
-    // call getStreamProvider, in addition to creating the provider, this method also adds values to the map that
-    // serialize the stream provider and make it possible to deserialize and recreate it for remote execution.
-    getStreamProvider( params );
     return invokeActionImpl( actionBean, actionUser, params );
   }
 
   /**
-   * Invokes the provided {@link IAction} locally as the provided {@code actionUser}.
+   * Invokes the provided {@link IAction} as the provided {@code actionUser}.
    *
    * @param actionBean the {@link IAction} being invoked
    * @param actionUser The user invoking the {@link IAction}
    * @param params     the {@link Map} or parameters needed to invoke the {@link IAction}
    * @return the {@link IActionInvokeStatus} object containing information about the action invocation
    * @throws Exception when the {@code IAction} cannot be invoked for some reason.
    */
-  protected IActionInvokeStatus invokeActionImpl( final IAction actionBean,
-                                                  final String actionUser,
-                                                  final Map<String, Serializable> params ) throws Exception {
+  protected final IActionInvokeStatus invokeActionImpl( final IAction actionBean,
```

Comment:
What is the reasoning behind making this method final?

Reply:
It's just my opinion that it should not be overridden.

Reply:
But what is behind this opinion?

Reply:
Gut feeling :) I can remove the keyword, if you feel strongly.

Reply:
Oh man, now I am in a dilema. On one hand I like listening to gut feelings. One the other hand reasoning says not to put a final ;-) Leave it up to you to decide...

Reply:
Basically - the public aip is invokeAction, and you can override it to your heart's desire, there's really no reason to override invokeActionImpl, which is an internal implementation, so my feeling is that if you are going to use this method, you should use it as-is. But again, just a gut feeling :)